<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="post_processing_manager_name" value="limelight_post_processing_manager"/>
    <!-- <include file="$(find tj2_limelight)/launch/limelight_targets.launch"/> -->
    
    <group ns="limelight">
        <node pkg="tj2_limelight" type="tj2_limelight" name="tj2_limelight" output="screen" respawn="true">
            <param name="video_url" value="http://10.0.88.42:5802/" />
            <param name="nt_host" value="10.0.88.2" />
            <param name="camera_info_url" value="package://tj2_limelight/config/320x240.yaml" />
            <!-- <param name="camera_info_url" value="package://tj2_limelight/config/960x720.yaml" /> -->
            <param name="frame_id" value="limelight_optical_link" />
            <param name="max_frame_rate" value="90.0" />
            <remap from="camera/image" to="color/image_raw" />
        </node>

        <node pkg="nodelet" type="nodelet" args="manager" name="$(arg post_processing_manager_name)" output="screen"/>

        <node pkg="nodelet" type="nodelet" name="register_depth"
                args="load depth_image_proc/register $(arg post_processing_manager_name)" respawn="false">
            <remap from="rgb/camera_info"             to="color/camera_info" />
            <remap from="depth/camera_info"           to="/camera/depth/camera_info" />
            <remap from="depth/image_rect"            to="/camera/depth/image_rect_raw" />
            <remap from="depth_registered/image_rect" to="aligned_depth_to_color/image_raw" />
        </node>
    </group>

    <node pkg="tf" type="static_transform_publisher" name="limelight_frame_to_optical" args="0.0 0.0 0.0  -0.500 0.500 -0.500 0.500 limelight_link limelight_optical_link 33" />
</launch>
